app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -1480,13 +1480,17 @@ BrowserWorld::CreateSkyBox(const std::string& aBasePath, const std::string& aExt
   ASSERT_ON_RENDER_THREAD();
   vrb::PausePerformanceMonitor pauseMonitor(*m.monitor);
   const bool empty = aBasePath == "cubemap/void";
+  if (empty) {
+    if (m.skybox) {
+      m.rootOpaqueParent->RemoveNode(*m.skybox->GetRoot());
+      m.skybox = nullptr;
```

Comment:
I believe this will leak the cube layer on devices that support async layers. I think you need to do:
```
  VRLayerCubePtr layer = m.skybox->GetLayer();
  if (layer) {
    m.device->DeleteLayer(layer);
  }
```
@MortimerGoro is that correct?

Also you can just call:
`m.skybox->RemoveFromParents();`

Reply:
yes, that's correct